provisioning/device/src/Transports/Amqp/AmqpClientConnection.cs

```diff
@@ -16,7 +16,7 @@
 
 namespace Microsoft.Azure.Devices.Provisioning.Client
 {
-    internal sealed class AmqpClientConnection : IDisposable
+    internal class AmqpClientConnection : IDisposable
```

Comment:
It appears that Moq doesn't allow to mock a sealed class, so changing here. Not sure this is the best approach so let me know your comments here.

Reply:
Try building Release (you should get a build error from SA), in which case you'll need to suppress the error (as I recently did [elsewhere](https://github.com/Azure/azure-iot-sdk-csharp/commit/ec484ea772f9ef2ae72a48ba50d07ccbaed1d570)).